In [1]:
#Si se usa Colab: 
!pip install beautifulsoup4
!pip install lxml

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [30]:
# Importamos la librería request para manejar las peticiones de acceso a la web
# y descarga de contenido.
import requests
# Especificamos la web
enlace = "https://en.wikipedia.org/wiki/Forbes%27_list_of_the_world%27s_highest-paid_athletes"

# Convertimos la web en texto
texto_web = requests.get(enlace).text


In [31]:
from bs4 import BeautifulSoup
sopa = BeautifulSoup(texto_web, 'lxml')


In [5]:
# Ahora vamos a ir trabajando con los distintos elementos de acceso a los datos.
# Título de la web:
print(sopa.title)
print(sopa.title.string)


<title>Forbes' list of the world's highest-paid athletes - Wikipedia</title>
Forbes' list of the world's highest-paid athletes - Wikipedia


In [6]:
print(sopa.href)

None


In [32]:
sopa.find_all("table")
hola = ""

In [11]:
# Si queremos quitar las etiquetas de HTML usamos la siguiente función:
print(sopa.title.string)

Forbes' list of the world's highest-paid athletes - Wikipedia


In [12]:
# Podemos ver los enlaces de la web a través del comando "sopa.a"
sopa.a

#Esto nos devuelve un único enlace, pero en la web podemos encontrar muchos...

<a id="top"></a>

In [35]:
# Para listar todos los enlaces usamos "sopa.find_all()"
sopa.find_all('a')
hola = ""

In [33]:
# La información que queremos se encuentra en las tablas, vamos a recuperarlas
tablas = sopa.find_all('table')


In [34]:
# Al existir varias tablas, tenemos que localizarlas.
# Para ello tenemos 2 opciones, ver todo el código o localizar manualmente 
# inspeccionando la página web.

# No todas las webs tienen la misma estructura siempre, por lo que para localizar 
# la información concreta, buscaremos inspeccionando en la web con F12.

# La clase de CSS de las tablas es "wikitable sortable"

tablas_listados_todos_anios = sopa.find_all('table', class_= 'wikitable sortable')


In [29]:
# Vamos a seleccionar la primera tabla, la de 2022
tabla_2022 = tablas_listados_todos_anios[0]


In [28]:
# Para localizar a los deportistas, 
# vamos a transformar la tabla a una estructura dataframe de pantas

import pandas as pd
dataframe_informacion_2022 = pd.read_html(str(tabla_2022))[0]
(dataframe_informacion_2022)
dataframe_informacion_total = pd.read_html(str(tablas_listados_todos_anios))


In [25]:
!pip install XlsxWriter



Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 149 kB 29.2 MB/s 


In [26]:
# Guardamos la información en el archivo:
archivo_informacion_22 = pd.ExcelWriter('deportistas_mejor_pagados_2022.xlsx', engine='xlsxwriter')
dataframe_informacion_2022.to_excel(archivo_informacion_22, sheet_name='List')
archivo_informacion_22.save()

In [38]:
#World's Billionaires
link = "https://en.wikipedia.org/wiki/The_World%27s_Billionaires"
web_text = requests.get(link).text


In [40]:
soup = BeautifulSoup(web_text, 'lxml')


In [57]:
link = "https://en.wikipedia.org/wiki/The_World%27s_Billionaires"
web_text = requests.get(link).text
soup = BeautifulSoup(web_text, 'lxml')

soup.find_all("table")
import numpy as np
tables_allyears = soup.find_all('table', class_= 'wikitable sortable')
table_2022_richest = tables_allyears[0]
table_2020_richest = tables_allyears[2]
dataframe_richest_2022 = pd.read_html(str(table_2022_richest))[0]
dataframe_richest_2020 = pd.read_html(str(table_2020_richest))[0]
dataframe_richest_2022.describe()
print("The mean age of the top 10 billionaires is "+ str(np.mean(dataframe_richest_2020["Age"]))+ " years old in 2020 according to Forbes.")



The mean age of the top 10 billionaires is 69.2 years old in 2020 according to Forbes.


In [69]:
#A qué se corresponde el número binario 01011001 en ASCII?
binario = "01011001"
int("010",2)
chr(87)


'W'

In [138]:
def binary_to_ASCII():
  loops = 1
  while loops >= 1:
    loops = 0
    bini = input("What number would you like to convert? \n")
    number = int(bini, 2)
    ASCII = chr(number)
    print("The character you selected is " + str(ASCII))
    tryagain = input("Would you like to try again? \n Yes or no \n")
    if tryagain == "yes":
      loops += 1
    else:
      print("See you again soon!")
      break

binary_to_ASCII()

What number would you like to convert? 
111111
The character you selected is ?
Would you like to try again? 
 Yes or no 
no
See you again soon!


In [165]:
import numpy as np



link = "https://proyectoa.com/tabla-de-correspondencias-ascii-y-iso-8859-1/"
web_text = requests.get(link).text
soup = BeautifulSoup(web_text, 'lxml')
tables_ascii = soup.find_all('table')

table_ascii1 = tables_ascii[0]
dataframe_ascii = pd.read_html(str(table_ascii1))[0]
hola = dataframe_ascii.describe()
print(type(dataframe_ascii))
tabla_ASCII = pd.read_html(link)[0]



print(tabla_ASCII[3].iloc[89+2])
#dataframe_richest_2022 = pd.read_html(str(table_2022_richest))[0]
#dataframe_richest_2022.describe()
#("The mean age of the top 10 billionaires is "+ str(np.mean(dataframe_richest_2020["Age"]))+ " years old in 2020 according to Forbes.")

<class 'pandas.core.frame.DataFrame'>
Y


In [143]:
def binary_to_ASCII():
  loops = 1
  while loops >= 1:
    loops = 0
    bini = input("What number would you like to convert? \n")
    number = int(bini, 2)
    ASCII = tabla_ASCII[3].iloc[number + 2]
    print("The character you selected is " + str(ASCII))
    tryagain = input("Would you like to try again? \n Yes or no \n")
    if tryagain == "yes":
      loops += 1
    else:
      print("See you again soon!")
      break

binary_to_ASCII()

What number would you like to convert? 
1111
The character you selected is DLE
Would you like to try again? 
 Yes or no 
yes
What number would you like to convert? 
01011001
The character you selected is Y
Would you like to try again? 
 Yes or no 
no
See you again soon!


In [168]:
link = "https://vandal.elespanol.com/guias/guia-league-of-legends-trucos-consejos-y-secretos/objetos"
webtext = requests.get(link).text
data_lol = pd.read_html(link)[0]
sopa = BeautifulSoup(webtext, "lxml")
tablas_lol = sopa.find_all("table", class_= "table t08")
tabla_oro = tablas_lol[1]
data_oro = pd.read_html(str(tabla_oro))[0]
nuevo_oro = data_oro.replace(300,3000)
print(data_oro.describe())
#import numpy as np
#tables_allyears = soup.find_all('table', class_= 'wikitable sortable')
#table_2022_richest = tables_allyears[0]
#table_2020_richest = tables_allyears[2]
#dataframe_richest_2022 = pd.read_html(str(table_2022_richest))[0]
#dataframe_richest_2020 = pd.read_html(str(table_2020_richest))[0]
#dataframe_richest_2022.describe()

       Coste de oro
count     17.000000
mean     580.000000
std      333.405721
min      150.000000
25%      350.000000
50%      450.000000
75%      700.000000
max     1300.000000


In [180]:
link = "https://vandal.elespanol.com/guias/guia-league-of-legends-trucos-consejos-y-secretos/objetos"
webtext = requests.get(link).text
data_lol = pd.read_html(link)[0]
sopa = BeautifulSoup(webtext, "lxml")
tablas_lol = sopa.find_all("table", class_= "table t08")
tabla_oro = tablas_lol[4]
data_oro = pd.read_html(str(tabla_oro))[0]
nuevos = data_oro.replace(to_replace = 300, value = 3000)
# print(data_oro)
print(nuevos.describe())

       Coste de oro
count     25.000000
mean    3045.320000
std      305.473774
min     2500.000000
25%     2800.000000
50%     3200.000000
75%     3300.000000
max     3400.000000


In [200]:
link = "https://leagueoflegends.fandom.com/es/wiki/Lista_del_da%C3%B1o_de_ataque_de_todos_los_campeones"
webtext = requests.get(link).text
data_lol = pd.read_html(link)[0]
sopa = BeautifulSoup(webtext, "lxml")
tablas_lol = sopa.find_all("table", class_= "wikitable sortable")
tabla_oro = tablas_lol
data_oro = pd.read_html(str(tabla_oro))[0]
#nuevos = data_oro.replace(to_replace = 300, value = 3000)
print(max(data_oro["Valor a nivel 18"]))
print(data_oro.loc[data_oro["Valor a nivel 18"] == 158])
#print(nuevos.describe())

158.0
   Campeón  Base  Aumento por nivel  Valor a nivel 1  Valor a nivel 18
39  Illaoi  68.0                5.0             73.0             158.0


In [202]:
!pip install texttable
import requests
from bs4 import BeautifulSoup
import texttable as texttable

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [211]:
url = "https://www.worldometers.info/coronavirus/countries-where-coronavirus-has-spread/"
webtext = requests.get(url).text
sopa = BeautifulSoup(webtext, "lxml")
tablas_covid = sopa.find_all("table", class_= "table-striped")
data_covid = pd.read_html(str(tablas_covid))[0]
print(data_covid)

            Country     Cases   Deaths             Region
0     United States  87321703  1035847      North America
1             India  43232004   524771               Asia
2            Brazil  31456865   668177      South America
3            France  29849919   148749             Europe
4           Germany  26847783   140183             Europe
..              ...       ...      ...                ...
224  Western Sahara        10        1             Africa
225      MS Zaandam         9        2                NaN
226            Niue         9        0  Australia/Oceania
227          Tuvalu         3        0  Australia/Oceania
228    Saint Helena         2        0             Africa

[229 rows x 4 columns]


In [214]:
max(data_covid["Deaths"])
data_covid[data_covid["Deaths"] == 1035847]


Country     Cases   Deaths         Region
0  United States  87321703  1035847  North America

In [220]:
from tabulate import tabulate
bonito = tabulate(data_covid.head(), tablefmt = "pipe", headers = "keys" )
print(bonito)

|    | Country       |    Cases |   Deaths | Region        |
|---:|:--------------|---------:|---------:|:--------------|
|  0 | United States | 87321703 |  1035847 | North America |
|  1 | India         | 43232004 |   524771 | Asia          |
|  2 | Brazil        | 31456865 |   668177 | South America |
|  3 | France        | 29849919 |   148749 | Europe        |
|  4 | Germany       | 26847783 |   140183 | Europe        |


|    | Country       |    Cases |   Deaths | Region        |
|---:|:--------------|---------:|---------:|:--------------|
|  0 | United States | 87321703 |  1035847 | North America |
|  1 | India         | 43232004 |   524771 | Asia          |
|  2 | Brazil        | 31456865 |   668177 | South America |
|  3 | France        | 29849919 |   148749 | Europe        |
|  4 | Germany       | 26847783 |   140183 | Europe        |


In [215]:
# Importamos la web
pagina_covid = requests.get(url)
sopa = BeautifulSoup(pagina_covid.text, 'html.parser')
# Creamos la variable para guardar los datos
datos_covid = []
# Ahora usemos iter().
# La función iter() crea un objeto que se puede iterar, recorrer.
# Estos objetos son útiles cuando se combinan con bucles como for loop, while loop.

# sopa.find_all('td') atrapará todos los enlaces de la tabla
datos_iterados = iter(sopa.findAll('td')) 
# iter recorrerá lña tabla de manera automática
# Este bucle seguirá repitiéndose hasta que haya datos disponibles en el iterador
while True:
    try:
        pais = next(datos_iterados).text
        num_casos_confirmados = next(datos_iterados).text
        num_muertes = next(datos_iterados).text
        continentes = next(datos_iterados).text
 
        # Para 'num_casos_confirmados' and 'num_muertes', 
        # nos aseguramos de eliminar las comas y luego las convertiremos a int
        datos_covid.append((
            pais,
            int(num_casos_confirmados.replace(',', '')),
            int(num_muertes.replace(',', '')),
            continentes
        ))
 
    # StopIteration: se genera un error cuando no quedan más elementos para iterar
    except StopIteration:
        break

In [246]:
url = "https://www.imdb.com/search/name/?birth_place=Spain&adult=include&count=100&start=1&ref_=rlm"
webtext = requests.get(url).text
sopa = BeautifulSoup(webtext, "lxml")
div_oscar = sopa.find_all("div", class_= "lister-item mode-detail")
(div_oscar[0])
nombres = []
nombre = div_oscar[0].find_all("a")[1].text

#data_oscar = pd.read_html(str(tabla_oscar))
#print(data_oscar)
#data_covid = pd.read_html(str(tablas_covid))[0]
#print(data_covid)

 Millie Bobby Brown



In [247]:
from IPython.core.display import HTML

# Convertir una URL en una iamgen
def path_to_image_html(path):
    return '<img src="'+ path + '" width="60" >'

pd.set_option('display.max_colwidth', -1)

# informacion_actores: Dataframe con la información
# Imagen: Columna con la URL de las imánes de los actores y actrices
HTML(informacion_actores.to_html(escape=False ,formatters=dict(Imagen=path_to_image_html)))

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:7: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  import sys


NameError: ignored

In [264]:
str(div_oscar[0].find_all('a')[0]).partition("src=")[2].partition(" ")





('"https://m.media-amazon.com/images/M/MV5BMjA5NzA0NzQzMF5BMl5BanBnXkFtZTgwMTQxNjUzNjM@._V1_UX140_CR0,0,140,209_AL_.jpg"',
 ' ',
 'width="140"/>\n</a>')